### DISCUSSION TITLE
如何对定理、引理、推论等数学环境一起编号？

### DISCUSSION THREAD

**Discussion by CalvinWaWang** [sjtug/SJTUThesis#778] [提问] [answered]
现在在每一章节里面 定理，引理，命题等等都是分别编号的。
如何使之编号为类似如下：
定理2.1
定理2.2
命题2.3
推论2.4
......
谢谢解答！

**Comment by LogCreative** [comment, marked as answer]
https://github.com/sjtug/SJTUThesis/blob/3436a7b82eb6d63f3085cb7545b84baf826e03e5/texmf/tex/latex/sjtuthesis/sjtuthesis.cls#L1189-L1203

这个地方定义了这些环境，[`ntheorem`](https://mirrors.sjtug.sjtu.edu.cn/CTAN/macros/latex/contrib/ntheorem/ntheorem.pdf) 和 [`amsthm`](https://mirrors.sjtug.sjtu.edu.cn/CTAN/macros/latex/required/amscls/doc/amsthdoc.pdf#3) 都提供了这样一种方法：先设定一个主要编号定理环境，比如 `theorem`，之后想要共同编号的定理环境在不同的位置使用可选参数指定计数器，比如：
```latex
  \newtheorem{theorem}    {\sjtu@name@theorem}    [chapter]
  \newtheorem{assumption} [theorem] {\sjtu@name@assumption}
```
直接修改文档类中的内容是可行的（你需要调整 `theorem` 的位置为第一个），在主文档覆盖加载也是可行的：
```latex
 % setup.tex 中添加
  \renewtheorem{corollary}[theorem]{引理}
```

**Reply by CalvinWaWang** [reply]
非常感谢，属于是真学到一点知识了，感谢！！！！！
